groups/bsl/bslstl/bslstl_variant.h

```diff
@@ -0,0 +1,3548 @@
+    // template parameter 'TYPE', will destroy that instance.
+
+  public:
+    // ACCESSORS
+    template <class TYPE>
+    void operator()(TYPE& object) const
+    {
+        bslma::DestructionUtil::destroy(&object);
+    }
+};
+                       // ==============================
+                       // struct Variant_LessThanVisitor
+                       // ==============================
+
+#define BSLSTL_VARIANT_RELOP_VISITOR_DEFINITON(NAME, OP)                      \
+    template <class VARIANT>                                                  \
+    struct Variant_##NAME##Visitor {                                          \
+        Variant_##NAME##Visitor(const VARIANT *variant)                       \
+        : d_variant_p(variant)                                                \
+        {                                                                     \
+            BSLS_ASSERT_SAFE(d_variant_p);                                    \
+        }                                                                     \
+                                                                              \
+      private:                                                                \
+        const VARIANT *d_variant_p;                                           \
+                                                                              \
+      public:                                                                 \
+        template <size_t INDEX, class TYPE>                                   \
+        bool operator()(bsl::in_place_index_t<INDEX>,                         \
+                        const TYPE&                  value) const             \
+        {                                                                     \
+            return (bsl::get<INDEX>(*d_variant_p) OP value);                  \
+        }                                                                     \
+    };                                                                        \
+                                                                              \
+    template <class VARIANT>                                                  \
+    bool Variant_ImpUtil<VARIANT>::NAME##Impl(const VARIANT& lhs,             \
+                                              const VARIANT& rhs)             \
+    {                                                                         \
+        BSLS_ASSERT(lhs.index() == rhs.index() &&                             \
+                    !lhs.valueless_by_exception());                           \
+                                                                              \
+        Variant_##NAME##Visitor<VARIANT> NAME##Visitor(                       \
+                                                   BSLS_UTIL_ADDRESSOF(lhs)); \
+        return BSLSTL_VARIANT_VISITID(bool, VARIANT, NAME##Visitor, rhs);     \
+    }
```

Comment:
Where do you want the comment: in the Visitor or where the visitor is used ?
There's also the unfortunate struct name comment. I'm not sure what to do there. `struct Variant_RELOPVisitor` ? Suggestions welcome.

Reply:
I think the comment should be in the visitor itself. If you look at the place where the visitor is used (the `Impl` function) it's already obvious which argument is the LHS and which is the RHS, because they're named `lhs` and `rhs`.

I'm not sure what you mean about the struct name. `Variant_##NAME##Visitor` seems fine to me; I don't see any particular reason to change it.

Reply:
bump